File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> freq;
        int majority = 0;
        for (auto n:nums) {
            if (freq.count(n)) freq[n]+=1;
            else freq[n]=1;

            if (freq[n] > nums.size()/2) {
                majority = n;
                break;
            }
        }

        return majority;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        freqdict = {}
        for i in range(n):
            if nums[i] not in freqdict:
                freqdict[nums[i]] = 1
            else:
                freqdict[nums[i]] += 1
            if freqdict[nums[i]] > n/2:
                return nums[i]
        